src/main/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -266,5 +268,100 @@ public String toString() {
                 " length:"+ getReferenceLength()+" "+
                 " md5:"+md5()+")";
     }
+
+    public static final List<String> DEFAULT_DICTIONARY_EQUAL_TAG = Arrays.asList(
+            SAMSequenceRecord.MD5_TAG,
+            SAMSequenceRecord.SEQUENCE_LENGTH_TAG);
+
+    /**
+     * Will merge dictionaryTags from two dictionaries into one focusing on merging the tags rather than the sequences.
+     *
+     * Requires that dictionaries have the same SAMSequence records in the same order.
+     * For each sequenceIndex, the union of the tags from both sequences will be added to the new sequence, mismatching
+     * values (for tags that are in both) will generate a warning, and the value from dict1 will be used.
+     * For tags that are in tagsToEquate. an unequal value will generate an error (an IllegalArgumentException will
+     * be thrown.)
+     *
+     * @param dict1 first dictionary
+     * @param dict2 first dictionary
+     * @param tagsToMatch list of tags that must be equal if present in both sequence. Typical values will be MD, and LN
+     * @return dictionary consisting of the same sequences as the two inputs with the merged values of tags.
+     */
+    static public SAMSequenceDictionary mergeDictionaries(final SAMSequenceDictionary dict1,
+                                                          final SAMSequenceDictionary dict2,
+                                                          final List<String> tagsToMatch) {
+
+        if(dict1.getSequences().size()!=dict2.getSequences().size()) {
+            throw new IllegalArgumentException(String.format("Do not use this function to merge dictionaries with " +
+                    "different number of sequences in them. Found %d and %d sequences",
+                    dict1.getSequences().size(), dict2.getSequences().size()));
+        }
+        final SAMSequenceDictionary finalDict = new SAMSequenceDictionary();
+        for (final SAMSequenceRecord sequence : dict1.getSequences()) {
+            final SAMSequenceRecord mergedSequence = new SAMSequenceRecord(sequence.getSequenceName(),
+                    UNKNOWN_SEQUENCE_LENGTH);
+            finalDict.addSequence(mergedSequence);
+            final int sequenceIndex = sequence.getSequenceIndex();
+            if (!dict2.getSequence(sequenceIndex).getSequenceName().equals(sequence.getSequenceName())) {
```

Comment:
I think it would be better to validate up front, and then here you can assume everything is going to line up.